From 1be37714b21f98b64c4385c431ce2287c3b9eae6 Mon Sep 17 00:00:00 2001
From: Vineesh k k <vineesh.k.k@intel.com>
Date: Fri, 16 Oct 2015 15:45:13 +0530
Subject: [PATCH] EM/Charger: Fix for host device does not work after waking 
 		from sleep.

OTG VBUS was disabled by WDT during suspend. Disabled
the WDT during OTG operation.

Change-Id: I82400312949f5632b64c66454246dd406b0461c7
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-1109
Signed-off-by: Vineesh k k <vineesh.k.k@intel.com>
---
 drivers/power/bq24192_charger.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/power/bq24192_charger.c b/drivers/power/bq24192_charger.c
index 07d02fec97ac..ad2d12431992 100644
--- a/drivers/power/bq24192_charger.c
+++ b/drivers/power/bq24192_charger.c
@@ -965,11 +965,11 @@ static int bq24192_turn_otg_vbus(struct bq24192_chip *chip, bool votg_on)
 			bq24192_clear_hiz(chip);
 			/* Program the timers */
 			ret = program_timers(chip,
-						CHRG_TIMER_EXP_CNTL_WDT80SEC,
+						CHRG_TIMER_EXP_CNTL_WDTDISABLE,
 						false);
 			if (ret < 0) {
 				dev_warn(&chip->client->dev,
-					"TIMER enable failed %s\n", __func__);
+					"TIMER disable failed %s\n", __func__);
 				goto i2c_write_fail;
 			}
 
@@ -993,9 +993,6 @@ static int bq24192_turn_otg_vbus(struct bq24192_chip *chip, bool votg_on)
 				goto i2c_write_fail;
 			}
 			chip->boost_mode = true;
-			/* Schedule the charger task worker now */
-			schedule_delayed_work(&chip->chrg_task_wrkr,
-						0);
 	} else {
 			/* Clear the charger from the OTG mode */
 			if ((chip->chip_type == BQ24296) ||
@@ -1023,8 +1020,6 @@ static int bq24192_turn_otg_vbus(struct bq24192_chip *chip, bool votg_on)
 				goto i2c_write_fail;
 			}
 			chip->boost_mode = false;
-			/* Cancel the charger task worker now */
-			cancel_delayed_work_sync(&chip->chrg_task_wrkr);
 	}
 
 	/*
-- 
2.6.0.rc0.2.g7662973

